### DISCUSSION TITLE
DatePicker State Out of Sync with Visible Calendar

### DISCUSSION THREAD

**Discussion by arkmech** [adobe/react-spectrum#4127] [Q&A]
I am following this example: https://codesandbox.io/s/immutable-haze-s2rkrp?file=/src/DatePicker.js

![datecomponent](https://user-images.githubusercontent.com/114366662/221391158-d080dfd5-d60c-4821-b135-c70a279583ef.png)

My implementation is different in regards to:
- Using Radix `<Popover />` Primitive
- Trigger is only a `<Button />` (Not using react-aria's useButton) because I don't think compatible with Radix <Popover /> Trigger

**What happens:**
- When I set the date to lets say Feb something
- Close Popover
- Open Popover (shows on Feb again, expected)
- Change month to Aug
- Close Popover
- Open Popover again, Shows Aug (aka the month you previous had open) - Not what I want.
- I expected the Popover to open up the Modal to the current state of the DatePicker (aka Month to Feb).
- I also tried manually setting the datepicker state or the calendar state and they do show the correct values, but regardless the state of the visible calendar is in a world of its own. 
- The `useCalendarState` does take in the `calendarProps` returned from useDatePicker`

**Expected:**
The example shows that when you change the value in the `<DateField />` and open the Popover again, it opens to the correct visible calendar month with the correct date selected.

Is there something obvious I'm missing in regards how the visible Calendar Month is shown? I thought it was dictated by the DatePicker State which in turns dictates the Calendar State (thought this effects the visible calendar month)

**Comment by LFDanLu** [comment]
> I thought it was dictated by the DatePicker State which in turns dictates the Calendar State (thought this effects the visible calendar month)

That is correct, the state from `useDatePickerState` is used by `useDatePicker` which returns a `calendarProps` that are consumed by `useCalendarState`. Included in those `calendarProps` is a controlled `value` that is equal to the tracked date value in `useDatePickerState`. Mind sharing your implementation? Is the state returned by `useCalendarState` not returning the same calendar date as the one tracked by `useDatePickerState`?
